modules/mobile/spec/workers/vaccines_updater_job_spec.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Mobile::V0::VaccinesUpdaterJob, type: :job do
+  before(:all) do
+    @original_cassette_dir = VCR.configure(&:cassette_library_dir)
+    VCR.configure { |c| c.cassette_library_dir = 'modules/mobile/spec/support/vcr_cassettes' }
+  end
+
+  after(:all) { VCR.configure { |c| c.cassette_library_dir = @original_cassette_dir } }
+
+  it 'creates records for all vaccines in the group_names xml' do
+    VCR.use_cassette('vaccines/group_names') do
+      VCR.use_cassette('vaccines/manufacturers') do
+        service = described_class.new
+        expect do
+          service.perform
+        end.to change { Mobile::V0::Vaccine.count }.from(0).to(3)
+
+        no_manufacturer = Mobile::V0::Vaccine.find_by(cvx_code: 1)
+        expect(no_manufacturer.group_name).to eq('DTAP')
+        covid_vaccine = Mobile::V0::Vaccine.find_by(cvx_code: 207)
+        expect(covid_vaccine.group_name).to eq('COVID-19')
+        non_covid_vaccine = Mobile::V0::Vaccine.find_by(cvx_code: 2)
+        expect(non_covid_vaccine.group_name).to eq('POLIO')
+      end
+    end
+  end
+
+  it 'sets manufacturer when manufacturer data is available and group name is COVID-19' do
```

Comment:
Same for this spec, every time `when` makes an appearance that's the context :)

Reply:
I hear you. I usually avoid contexts unless there are multiple tests within it and perhaps data setup required for all tests within in. But I'm happy to conform to the shared style.